src/NamespaceUsages.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types = 1);
+
+namespace Spaze\PHPStan\Rules\Disallowed;
+
+use PhpParser\Node;
+use PhpParser\Node\Name\FullyQualified;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\ClassNotFoundException;
+use PHPStan\Rules\Rule;
+
+/**
+ * Reports on usages of classes in general.
+ *
+ * @package Spaze\PHPStan\Rules\Disallowed
+ * @implements Rule<Node>
+ */
+class NamespaceUsages implements Rule
+{
+
+	/** @var DisallowedNamespaceHelper */
+	private $disallowedHelper;
+
+	/** @var DisallowedNamespace[] */
+	private $disallowedNamespace;
+
+
+	/**
+	 * @param DisallowedNamespaceHelper $disallowedNamespaceHelper
+	 * @param array<array{namespace:string, message?:string, allowIn?:string[]}> $forbiddenNamespaces
+	 */
+	public function __construct(DisallowedNamespaceHelper $disallowedNamespaceHelper, array $forbiddenNamespaces)
+	{
+		$this->disallowedHelper  = $disallowedNamespaceHelper;
+		$this->disallowedNamespace = $this->disallowedHelper->createDisallowedNamespacesFromConfig($forbiddenNamespaces);
+	}
+
+
+	public function getNodeType(): string
+	{
+		return Node::class;
+	}
+
+
+	/**
+	 * @param Node $node
+	 * @param Scope $scope
+	 * @return string[]
+	 * @throws ClassNotFoundException
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if ($node instanceof FullyQualified) {
+			$namespace = $node->toString();
+		} elseif ($node instanceof Node\Stmt\UseUse) {
+			$namespace = $node->name->toString();
+		} elseif ($node instanceof Node\Expr\StaticCall && $node->class instanceof Node\Name) {
+			$namespace = $node->class->toString();
+		} else {
+			return [];
+		}
```

Comment:
Can you think of more types that are missing?

Reply:
There's a list of things that should be supported in #50, please add tests for all of them.

Reply:
Great, I think I've added all cases now :) 

Except `$this->namespacedObj->call)` (and `$this->x();` that is inherited) as I don't know how to do this. 

Is that really needed?